plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -629,6 +629,41 @@ public function backfill_post_record( $order ) {
 		do_action( 'woocommerce_hpos_post_record_backfilled', $order );
 	}
 
+	/**
+	 * Updates an order (in this datastore) from another order object.
+	 *
+	 * @param \WC_Abstract_Order $order Source order.
+	 * @return bool Whether the order was updated.
```

Comment:
Is there ever a situation where we don't return `true`? If not -- should there be? Or will there be one in the future? When only ever return true the return value feels a bit misleading, as a caller of the method might believe based on the return value that it can fail.

Reply:
So this docblock was just copied over from the parent class, which does return a value [based on whether the post was updated or not](https://github.com/woocommerce/woocommerce/blob/faa184e25126d975a38327e12f04e89fa0f18a19/plugins/woocommerce/includes/data-stores/abstract-wc-order-data-store-cpt.php#L680). That said, it's a bit misleading too since the metadata update could've failed (which in that case is actually all order data except status/date) and that's not taken into account.

Similarly, `save_meta_data()` doesn't tell us whether the meta was updated, so I'd say that right now we can't really tell whether the order was correctly updated or not, so the bool seems useless but we might eventually provided we make some adjustments elsewhere.